velox/exec/fuzzer/AggregationFuzzer.cpp

```diff
@@ -915,6 +928,114 @@ bool AggregationFuzzer::verifySortedAggregation(
   return resultOrError.exceptionPtr != nullptr;
 }
 
+bool AggregationFuzzer::verifyDistinctAggregation(
+    const std::vector<std::string>& groupingKeys,
+    const std::vector<std::string>& aggregates,
+    const std::vector<std::string>& masks,
+    const std::vector<RowVectorPtr>& input,
+    bool customVerification) {
+  auto firstPlan = PlanBuilder()
+                       .values(input)
+                       .singleAggregation(groupingKeys, aggregates, masks)
+                       .planNode();
+
+  std::vector<std::shared_ptr<ResultVerifier>> customVerifiers;
+  if (customVerification) {
+    const auto& aggregationNode =
+        std::dynamic_pointer_cast<const core::AggregationNode>(firstPlan);
+
+    for (auto i = 0; i < aggregates.size(); ++i) {
+      const auto& aggregate = aggregates.at(i);
+      if (customVerificationFunctions_.find(aggregate) ==
+          customVerificationFunctions_.end()) {
+        continue;
+      }
+
+      const auto& verifier = customVerificationFunctions_.at(aggregate);
+      verifier->initialize(
+          input,
+          groupingKeys,
+          aggregationNode->aggregates()[i],
+          aggregationNode->aggregateNames()[i]);
+      customVerifiers.push_back(verifier);
+    }
+  }
+
+  SCOPE_EXIT {
+    if (customVerification) {
+      resetCustomVerifiers(customVerifiers);
+    }
+  };
+
+  // Create all the plans upfront.
+  std::vector<PlanWithSplits> plans;
+  auto directory = exec::test::TempDirectoryPath::create();
+  const auto inputRowType = asRowType(input[0]->type());
+
+  if (isTableScanSupported(inputRowType)) {
+    directory = exec::test::TempDirectoryPath::create();
+    auto splits = makeSplits(input, directory->path);
+    plans.push_back(
+        {PlanBuilder()
+             .tableScan(inputRowType)
+             .singleAggregation(groupingKeys, aggregates, masks)
+             .planNode(),
+         splits});
+  }
+
+  if (persistAndRunOnce_) {
+    persistReproInfo(plans, reproPersistPath_);
+  }
+
+  try {
+    auto resultOrError = execute(firstPlan);
+    if (resultOrError.exceptionPtr) {
+      ++stats_.numFailed;
+    }
+
+    // TODO Use ResultVerifier::compare API to compare Velox results with
```

Comment:
Is this a copy-paste from somewhere? If so, would it be possible to refactor the code to avoid so much copy-paste?